x-pack/plugins/siem/public/components/open_timeline/note_previews/index.test.tsx

```diff
@@ -0,0 +1,144 @@
+
+    hasNotes[0].notes!.forEach(({ savedObjectId }) => {
+      expect(wrapper.find(`[data-test-subj="note-preview-${savedObjectId}"]`).exists()).toBe(true);
+    });
+  });
+
+  test('it does NOT render the preview container if notes is undefined', () => {
+    const wrapper = mountWithIntl(<NotePreviews />);
+
+    expect(wrapper.find('[data-test-subj="note-previews-container"]').exists()).toBe(false);
+  });
+
+  test('it does NOT render the preview container if notes is null', () => {
+    const wrapper = mountWithIntl(<NotePreviews notes={null} />);
+
+    expect(wrapper.find('[data-test-subj="note-previews-container"]').exists()).toBe(false);
+  });
+
+  test('it does NOT render the preview container if notes is empty', () => {
+    const wrapper = mountWithIntl(<NotePreviews notes={[]} />);
+
+    expect(wrapper.find('[data-test-subj="note-previews-container"]').exists()).toBe(false);
+  });
+
+  test('it filters-out non-unique savedObjectIds', () => {
+    const nonUniqueNotes: TimelineResultNote[] = [
+      {
+        note: 'a',
+        savedObjectId: 'noteId1',
+        updated: 1553400753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'b (savedObjectId is the same as the previous entry)',
+        savedObjectId: 'noteId1',
+        updated: 1553500753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'c',
+        savedObjectId: 'noteId2',
+        updated: 1553600753 * 1000,
+        updatedBy: 'bob',
+      },
+    ];
+
+    const wrapper = mountWithIntl(<NotePreviews notes={nonUniqueNotes} />);
+
+    expect(
+      wrapper
+        .find(`[data-test-subj="updated-by"]`)
+        .at(2)
+        .text()
+    ).toEqual('bob');
+  });
+
+  test('it filters-out null savedObjectIds', () => {
+    const nonUniqueNotes: TimelineResultNote[] = [
+      {
+        note: 'a',
+        savedObjectId: 'noteId1',
+        updated: 1553400753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'b (savedObjectId is null)',
+        savedObjectId: null,
+        updated: 1553500753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'c',
+        savedObjectId: 'noteId2',
+        updated: 1553600753 * 1000,
+        updatedBy: 'bob',
+      },
+    ];
+
+    const wrapper = mountWithIntl(<NotePreviews notes={nonUniqueNotes} />);
+
+    expect(
+      wrapper
+        .find(`[data-test-subj="updated-by"]`)
+        .at(2)
+        .text()
+    ).toEqual('bob');
+  });
+
+  test('it filters-out undefined savedObjectIds', () => {
+    const nonUniqueNotes: TimelineResultNote[] = [
+      {
+        note: 'a',
+        savedObjectId: 'noteId1',
+        updated: 1553400753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'b (savedObjectId is undefined)',
+        updated: 1553500753 * 1000,
+        updatedBy: 'alice',
+      },
+      {
+        note: 'c',
+        savedObjectId: 'noteId2',
+        updated: 1553600753 * 1000,
```

Comment:
It's difficult to read Epoch to compare these numbers numbers to see what is different.

Optional suggestion:

Use Momentjs here for these tests...Something like:
```ts
moment('1970-01-18T23:31:40.753Z').add(1, 'second')
```